src/hyperswarm-dht-logger.js

```diff
@@ -50,25 +52,29 @@ class HyperSwarmDHTLogger {
 
     this.fileWatcher = new Tail(this.file)
     this.fileWatcher.on('line', (line) => {
-      this.sockets.forEach(socket => {
+      Array.from(this.sockets.values()).forEach(socket => {
         socket.write(line)
       })
     })
     this.fileWatcher.watch()
   }
 
-  addSocket (socket) {
-    this.sockets.push(socket)
+  addSocket (id, socket) {
+    this.sockets.set(id, socket)
   }
 
-  removeSocket (socket) {
-    this.sockets.splice(this.sockets.indexOf(socket), 1)
-    socket.end()
-    socket.destroy()
+  removeSocket (id) {
+    const socket = this.sockets.get(id)
+
+    if (socket) {
+      this.sockets.delete(id)
+      socket.end()
+      socket.destroy()
+    }
   }
 
   async stop () {
-    this.sockets.slice().forEach(socket => this.removeSocket(socket))
+    Array.from(this.sockets.keys()).forEach(id => this.removeSocket(id))
```

Comment:
good usage here, this.removeSocket will remove item from map and that could possibly cause unexpected behavior, and since we added Array.from it's a new object that is not dependent on map

Reply:
How can this cause unexpected behavior?

Reply:
yep correct, that's why i said good usage here :)